# Data exploration – VIP Medical Group

This notebook explores the raw datasets provided for the technical assessment.
The goal is to understand data structure, data quality issues, and inconsistencies
that informed the design of the ETL pipeline.


In [9]:
import pandas as pd
from pathlib import Path

In [23]:
BASE_DIR = Path.cwd()
DATA_DIR = BASE_DIR / "data"
RAW_DATA_DIR = DATA_DIR / "raw"

APPOINTMENTS_PATH = RAW_DATA_DIR / "Data Engineer's Appointments Excel - VIP Medical Group.xlsx"
DOCTORS_PATH = RAW_DATA_DIR / "Data Enginner's Doctors Excel - VIP Medical Group.xlsx"

df_appointments = pd.read_excel(APPOINTMENTS_PATH)
df_doctors = pd.read_excel(DOCTORS_PATH)

In [11]:
df_appointments.head()

,booking_id,patient_id,doctor_id,booking_date,status
0,1,10.0,100.0,10/20/2025,confirmed
1,2,11.0,101.0,10/21/2025,cancelled
2,3,12.0,100.0,10/20/2025,confirmed
3,4,13.0,101.0,10/20/2025,confirmed
4,5,14.0,102.0,10/20/2025,cancelled


In [12]:
df_doctors.head()

,doctor_id,name,specialty
0,100,Dr. Pérez,Vein
1,101,Dr. Gómez,Pain
2,102,Dr. Sánchez,Vein
3,103,Dr. Rodríguez,Pain
4,104,Dr. Martínez,Vein


In [13]:
# Revisión básica
print("\n--- Información general ---")
print(df_appointments.info())
print("\n--- Dimensiones ---")
print(f"Filas: {df_appointments.shape[0]}, Columnas: {df_appointments.shape[1]}")
print("\n--- Nombres de columnas ---")
print(df_appointments.columns.tolist())


--- Información general ---
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1025 entries, 0 to 1024
Data columns (total 5 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   booking_id    1025 non-null   object 
 1   patient_id    1022 non-null   float64
 2   doctor_id     1025 non-null   float64
 3   booking_date  1025 non-null   object 
 4   status        1025 non-null   object 
dtypes: float64(2), object(3)
memory usage: 40.2+ KB
None

--- Dimensiones ---
Filas: 1025, Columnas: 5

--- Nombres de columnas ---
['booking_id', 'patient_id', 'doctor_id', 'booking_date', 'status']


In [14]:
# Revisión básica
print("\n--- Información general ---")
print(df_doctors.info())
print("\n--- Dimensiones ---")
print(f"Filas: {df_doctors.shape[0]}, Columnas: {df_doctors.shape[1]}")
print("\n--- Nombres de columnas ---")
print(df_doctors.columns.tolist())


--- Información general ---
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5 entries, 0 to 4
Data columns (total 3 columns):
 #   Column     Non-Null Count  Dtype 
---  ------     --------------  ----- 
 0   doctor_id  5 non-null      int64 
 1   name       5 non-null      object
 2   specialty  5 non-null      object
dtypes: int64(1), object(2)
memory usage: 252.0+ bytes
None

--- Dimensiones ---
Filas: 5, Columnas: 3

--- Nombres de columnas ---
['doctor_id', 'name', 'specialty']


In [15]:
# Estadísticas descriptivas
display(df_appointments.describe())
display(df_appointments.describe(include='object'))

,patient_id,doctor_id
count,1022.000000,1025.000000
mean,33.665362,102.443902
std,14.073389,1.761964
min,10.000000,100.000000
25%,22.000000,101.000000
50%,34.000000,102.000000
75%,46.000000,104.000000
max,58.000000,105.000000


,booking_id,booking_date,status
count,1025,1025,1025
unique,1025,339,7
top,1,2025-10-23,confirmed
freq,1,254,807


In [16]:
# Estadísticas descriptivas
display(df_doctors.describe())
display(df_doctors.describe(include='object'))

,doctor_id
count,5.000000
mean,102.000000
std,1.581139
min,100.000000
25%,101.000000
50%,102.000000
75%,103.000000
max,104.000000


,name,specialty
count,5,5
unique,5,2
top,Dr. Pérez,Vein
freq,1,3


In [17]:
# Valores nulos y duplicados
print("\n--- Valores nulos ---")
print(df_appointments.isnull().sum())
print("\n--- Duplicados ---")
print(df_appointments.duplicated().sum())


--- Valores nulos ---
booking_id      0
patient_id      3
doctor_id       0
booking_date    0
status          0
dtype: int64

--- Duplicados ---
0


In [18]:
# Valores nulos y duplicados
print("\n--- Valores nulos ---")
print(df_doctors.isnull().sum())
print("\n--- Duplicados ---")
print(df_doctors.duplicated().sum())


--- Valores nulos ---
doctor_id    0
name         0
specialty    0
dtype: int64

--- Duplicados ---
0


In [19]:
print("\n--- valores unicos por columna ---")
print(df_appointments.nunique())


--- valores unicos por columna ---
booking_id      1025
patient_id        49
doctor_id          6
booking_date     339
status             7
dtype: int64


In [20]:
print("\n--- valores unicos por columna ---")
print(df_doctors.nunique())


--- valores unicos por columna ---
doctor_id    5
name         5
specialty    2
dtype: int64


In [21]:

set(df_appointments['doctor_id'].dropna().astype(int)) - set(df_doctors['doctor_id'])

{105}

In [22]:
df_appointments['status'].value_counts()

status
confirmed     807
cancelled     209
confirmed       5
canceled        1
Confirmed       1
confirmed.      1
cancelled       1
Name: count, dtype: int64

## Key Findings

- `booking_id` contains non-numeric values (e.g., '37X')
- Appointment status values are inconsistent in casing and spelling
- A doctor_id (105) exists in appointments but not in the doctors dataset
- A significant number of booking_date values are invalid or inconsistent
- No duplicated rows were found in either dataset
